### PR TITLE
Add check-in restriction attributes and checks

### LINKED ISSUE #4856: Allow restricting check-in and auto-check-in for certain ticket types
**Is your feature request related to a problem? Please describe.**
The API should be able to restrict check in for all attendees holding certain ticket types

**Describe the solution you'd like**

There should be an attribute of for Ticket API called as 'is-check-in-restricted`, and 'auto-check-in-enabled'.

**Additional context**

Orga app issues: [#973](https://github.com/fossasia/open-event-orga-app/issues/973) [#974](https://github.com/fossasia/open-event-orga-app/issues/974)

### PR DESCRIPTION
**Pull request by schedutron**
Fixes #4856

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
This PR adds the check-in restriction attributes in the ticket API, to signal whether check-in is restricted and whether automatic check-in is allowed.

### CODE REVIEW

**Comment by schedutron** [comment]
I'm working on adding the checks as well.

**Comment by srv-twry** [comment]
> I'm working on adding the checks as well.

Please add WIP to the PR title

**Comment by mishuvs** [comment]
@schedutron please fix the failing build

**Comment by srv-twry** [comment]
@schedutron The build is failing due to multiple heads in the db. Update your migration file's `down_revision` to the current head now.

File: docs/api/api_blueprint.apib [hunk 1]
```diff
@@ -13682,6 +13682,8 @@ Create a new ticket using an event_id.
                         "is-hidden": "false",
                         "min-order": "1",
                         "max-order": "100"
+                        "is-check-in-restricted": "true",
+                        "auto-check-in-enabled": "false"
```
Comment (srv-twry) [inline docs/api/api_blueprint.apib]: You have to add it in all the instances of tickets. For eg. in the GET ticket, GET list of tickets, PATCH tickets, GET tickets under an event etc. etc.

File: migrations/versions/6440077182f0_.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+"""
+
+from alembic import op
+import sqlalchemy as sa
+# import sqlalchemy_utils
+
+
+# revision identifiers, used by Alembic.
+revision = '6440077182f0'
+down_revision = '4bbba854e30b'
```
Comment (srv-twry) [inline migrations/versions/6440077182f0_.py]: build fails again :disappointed: . Update it again, I know it's annoying.

**Comment by mishuvs** [comment]
@schedutron This has been WIP for quite some time. Please complete this PR as it is blocking a PR in the Orga App.

**Comment by schedutron** [comment]
@mishuvs I have added the attributes, but I'm not sure about what kind of checks have to be added for these attributes. Can you please guide me for the same?

**Comment by mishuvs** [comment]
@schedutron I am not sure about what checks need to be included.
@iamareebjamal @srv-twry @sridharjajoo @Masquerade0097 Can anyone help?

**Comment by mishuvs** [comment]
@schedutron maybe we can add the checks later if required.

**Comment by schedutron** [comment]
@srv-twry @mayank8318 @iamareebjamal Please review :)

**Comment by srv-twry** [comment]
@schedutron I triggered a rebuild and it failed with multiple migration heads. Please update your migrations file.

**Comment by schedutron** [comment]
@srv-twry @iamareebjamal @mayank8318 @bhaveshAn Please review, it's a blocker for the orga app.

**Comment by srv-twry** [comment]
@schedutron Please squash the commits.

**Comment by srv-twry** [comment]
@schedutron Please fix the build.

**Pull request merged**